### PR TITLE
tab expansion not working correctly for some parameters

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
code was indexing outside of it's range for the last element causing an exception which fails the whole discovery operation

addresses #1345 

need some pointers on how to add a test for this

### CODE REVIEW

**Review by lzybkr — CHANGES_REQUESTED** [review]
Please add a test.

**Comment by SteveL-MSFT** [comment]
Didn't find any existing tests nor tests for TabExpansion++ so not sure how to validate

**Comment by lzybkr** [comment]
You don't need TabExpansionPlusPlus.

You can look at:

```
#212 PS> rg TabExpansion2
powershell\Host\TabCompletion\BugFix.Tests.ps1
4:        $result = TabExpansion2 -inputScript "[system.manage" -cursorColumn "[system.manage".Length
11:        $result = TabExpansion2 -inputScript "1 -sp" -cursorColumn "1 -sp".Length
18:        $result = TabExpansion2 -inputScript "1 -a" -cursorColumn "1 -a".Length
25:        $result = TabExpansion2 -inputScript "[pscu" -cursorColumn "[pscu".Length
#213 PS> rg CompleteInput
powershell\Language\CompletionTestSupport.psm1
69:    $results = [System.Management.Automation.CommandCompletion]::CompleteInput(

powershell\Language\Classes\Scripting.Classes.BasicParsing.Tests.ps1
613:        $completions = [System.Management.Automation.CommandCompletion]::CompleteInput($line, $line.Length, $null)

powershell\Language\Parser\ExtensibleCompletion.Tests.ps1
63:    $results = [System.Management.Automation.CommandCompletion]::CompleteInput(
```

**Comment by SteveL-MSFT** [comment]
Test added

File: test/powershell/Host/TabCompletion/BugFix.Tests.ps1 [hunk 1]
```diff
@@ -27,4 +27,13 @@ Describe "Tab completion bug fix" -Tags "CI" {
         $result.CompletionMatches.Count | Should Be 1
         $result.CompletionMatches[0].CompletionText | Should Be "pscustomobject"
     }
+    It "Issue#1345 - 'Import-Module -n<tab>' should work" {
+        $cmd = "Import-Module -n"
+        $result = TabExpansion2 -inputScript $cmd -cursorColumn $cmd.Length
+        $result | Should Not BeNullOrEmpty
```
Comment (iSazonov) [inline test/powershell/Host/TabCompletion/BugFix.Tests.ps1]: If below we check `count` we should remove the line.

Comment (SteveL-MSFT) [inline test/powershell/Host/TabCompletion/BugFix.Tests.ps1]: Agree it's unnecessary

File: test/powershell/Host/TabCompletion/BugFix.Tests.ps1 [hunk 2]
```diff
@@ -27,4 +27,12 @@ Describe "Tab completion bug fix" -Tags "CI" {
         $result.CompletionMatches.Count | Should Be 1
         $result.CompletionMatches[0].CompletionText | Should Be "pscustomobject"
     }
+    It "Issue#1345 - 'Import-Module -n<tab>' should work" {
```
Comment (iSazonov) [inline test/powershell/Host/TabCompletion/BugFix.Tests.ps1]: Is it best practice to place a issue number in `It` title? Maybe make a comment below?

Comment (SteveL-MSFT) [inline test/powershell/Host/TabCompletion/BugFix.Tests.ps1]: Currently just following convention already defined in this script.  Created new issue to get this test script cleaned-up: https://github.com/PowerShell/PowerShell/issues/3369

Comment (iSazonov) [inline test/powershell/Host/TabCompletion/BugFix.Tests.ps1]: 👍 
Closed.

**Comment by SteveL-MSFT** [comment]
@lzybkr anything else needed?

**Review by lzybkr — APPROVED** [review]

**Pull request merged**